src/android/toga_android/widgets/base.py

```diff
@@ -41,12 +61,12 @@ def set_enabled(self, value):
     # APPLICATOR
 
     def set_bounds(self, x, y, width, height):
-        # No implementation required here; the new sizing will be picked up
-        # by the container layout.
-        pass
+        if self.container:
+            # Ask the container widget to set our bounds.
+            self.container.set_child_bounds(self, x, y, width, height)
```

Comment:
Is there any particular reason to defer this to the box/container, rather than handling this directly on the base widget?

Reply:
The basic idea is that only `Box` knows that `Box` is a `RelativeLayout` instance, and `RelativeLayout__LayoutParams` is specific to `RelativeLayout`, so I am scared of referring to `RelativeLayout__LayoutParams` anywhere except `Box`.

More context: The tutorial code:

- Creates a Box
- Adds widgets to the Box
- Sets the window's `.content` to the Box
- Shows the window

So I figure the Box is the real `container`.

It's up for the debate, but that's the idea.

Reply:
That will be true... right up until Box isn't the only container. e.g., an OptionContainer or ScrollContainer can (as the names suggest) be the top level container. So, the decision is essentially whether BaseWidget knows about RelativeLayout, or whether every child widget  that can act as a container does.

Reply:
`ScrollContainer` will presumably be an Android `ScrollView` widget, so it won't be a `RelativeLayout`. That suggests to me that the path I've chosen is simplest and best. Curious for your take.

Reply:
Hrm... you may be right...